backend/api/publishing.py

```
@@ -107,61 +311,234 @@ async def save_scenario_draft(
+            # Set individual completion fields based on their actual completion state
+            scenario.name_completed = completion_status.get("name_completed", False)
+            scenario.description_completed = completion_status.get("description_completed", False)
+            scenario.student_role_completed = completion_status.get("student_role_completed", False)
+            scenario.personas_completed = completion_status.get("personas_completed", False)
+            scenario.scenes_completed = completion_status.get("scenes_completed", False)
+            scenario.images_completed = completion_status.get("images_completed", False)
+            scenario.learning_outcomes_completed = completion_status.get("learning_outcomes_completed", False)
+            scenario.ai_enhancement_completed = completion_status.get("ai_enhancement_completed", False)
+            scenario.grading_config_completed = completion_status.get("grading_config_completed", False)
+            
             scenario.updated_at = datetime.utcnow()
```

Ruff (0.14.1)

`datetime.datetime.utcnow()` used

(DTZ003)